ci/authn-gce/get_gce_id_tokens.sh

```diff
@@ -0,0 +1,165 @@
+# 2. Verifies the GCE instance exists and is in status running
+# 3. Executes a ssh curl command and write the output token with the appropriate token name to '../ci/authn-gce/tokens'.
+#	---------------------------------------------------------------------------------------------------------------------
+
+PROGNAME=$(basename "$0")
+INSTANCE_ZONE=us-central1-a
+TOKENS_OUT_DIR_PATH=../ci/authn-gce/tokens
+INSTANCE_EXISTS=0
+INSTANCE_RUNNING=0
+
+
+main() {
+  echo "-- ------------------------------------------ --"
+  echo "-- Generate Google Cloud GCE Identity tokens --"
+  echo "-- ------------------------------------------ --"
+
+  echo "-- Verifying 'gcloud' is installed..."
+  ensure_gcloud_is_installed
+  echo "-- Verifying CGE instance name..."
+  check_instance_name_arg "$1"
+  echo "-- Checking if GCE instance: '${INSTANCE_NAME}' exists..."
+  check_if_instance_exists_and_running "$INSTANCE_NAME"
+  echo "-- Generate tokens and writing to files under '${TOKENS_OUT_DIR_PATH}'..."
+  get_tokens_into_files
+  echo "-- Done."
+}
+
+ensure_gcloud_is_installed() {
+  COMMAND=gcloud
+
+  if ! command -v $COMMAND &> /dev/null; then
+    error_exit "-- ${COMMAND} could not be found."
+  fi
+  echo "-- ${COMMAND} command exists"
+}
+
+check_instance_name_arg() {
+  if [ -z "$1" ]; then
+    echo "-- GCE instance name is required. Usage: ./${PROGNAME} [GCE_INSTANCE_NAME]"
+    print_running_gce_instances
+    error_exit "GCE instance name is required!"
+  else
+    INSTANCE_NAME=$1
+    echo "-- GCE instance name is set to: '${INSTANCE_NAME}'."
+  fi
+}
+
+#	----------------------------------------------------------------
+# Checks if the given instance name argument exist and in RUNNING
+# status; exit the script with error otherwise.
+#	----------------------------------------------------------------
+check_if_instance_exists_and_running() {
+  local instance_name="$1"
+  # 'gcloud' response format: 'instance_zone;status', (e.g. cucumber-test;europe-west3-c).
+  local format="value[separator=';'](zone.basename(),status)"
+  # Filter list of al GCE instances by $instance_name. Should return 'instance_zone;status'
+  # if the instance exists; empty string otherwise.
+  local instance_info=$(gcloud compute instances list --filter="$instance_name" --format="$format")
+
+  if [[ "$instance_info" != "" ]]; then
+    INSTANCE_EXISTS=1
+    # Holds the instance status
+    local status="TERMINATED"
+    # Store IFS to restore later
+    local stored_ifs="$IFS"
+    # Export instance info delimiter
+    export IFS=";"
+    # Init for loop index
+    local index=0
+
+    #	----------------------------------------------------------------
+    # The $instance_info is in the form of 'instance_zone;status',
+    # (e.g. cucumber-test;europe-west3-c).
+    # The below for loop extract the following:
+    # - instance zone: used in the 'gcloud compute ssh' command.
+    # - instance status: used to validate that the instance is running.
```

Comment:
can't we split the string by `;` and set the values?

Reply:
instead of for? yes I had more tokens in the beginning. I'll try that.